### ISSUE TITLE
solver 参数 如何选择？

### ISSUE THREAD

**Issue by huhuhang** [shiyanlou/louplus-ml#118]
- 问题链接：https://www.shiyanlou.com/questions/165199
- 学员昵称：书山压力大EE
- 提问时间：2019-05-13T08:43:27.000Z

---

```
from sklearn.linear_model import LogisticRegression

model = LogisticRegression(solver='lbfgs')  # 定义逻辑回归模型
model.fit(X_train, y_train)  # 使用训练数据完成模型训练
```
solver的解读：
```
solver : str, {'newton-cg', 'lbfgs', 'liblinear', 'sag', 'saga'},              default: 'liblinear'.

    Algorithm to use in the optimization problem.

    - For small datasets, 'liblinear' is a good choice, whereas 'sag' and
      'saga' are faster for large ones.
    - For multiclass problems, only 'newton-cg', 'sag', 'saga' and 'lbfgs'
      handle multinomial loss; 'liblinear' is limited to one-versus-rest
      schemes.
    - 'newton-cg', 'lbfgs' and 'sag' only handle L2 penalty, whereas
      'liblinear' and 'saga' handle L1 penalty.

```
---

- multinomial loss  和one-versus-rest schemes   是什么意思？    

- 能不能把'newton-cg', 'lbfgs', 'liblinear', 'sag', 'saga'这几个方法的意思和使用情景阐述一下？

**Comment by songyer** [comment]
已回答
